app/views/shop/_messages.html.haml

```diff
@@ -10,7 +10,7 @@
              register: ('<a auth="signup">' + t('.register') + '</a>').html_safe}
         - else
           = t '.require_customer_html',
-            {contact: ('<a href="##contact">' + t('.contact') + '</a>').html_safe,
+            {contact: (link_to t('.contact'), "#/contact", method: :get),
```

Comment:
Maybe we can use Jquery to fire off the click event for the tab, something like this: 
`$('#tab_contact').find('a').click()`
That works for me in the console at least. Would avoid a page refresh (which is quite slow so I think worth avoiding). The problem I can see is that it would do nothing if it can't find the tab - I think it's ok as the _messages partial is always displayed with the shop details, but it's future-proof (e.g. if this message partial is reused on another page).

I'm not sure it's less hacky than Lynn's method but it is snappier :)